internal/controller/imagerepository_controller.go

```diff
@@ -877,3 +898,233 @@ func (r *ImageRepositoryReconciler) ImageRepositoryForSameUrlExists(ctx context.
 func getComponentSaName(componentName string) string {
 	return fmt.Sprintf("%s%s", componentSaNamePrefix, componentName)
 }
+
+// addPullSecretToApplicationPullSecret updates the application pull secret when new image repository pull secret is created
+// or when an existing one is updated.
+func (r *ImageRepositoryReconciler) addPullSecretToApplicationPullSecret(ctx context.Context, applicationName, namespace, pullSecretName string, overwrite bool) error {
+	log := ctrllog.FromContext(ctx)
+
+	application := &appstudioredhatcomv1alpha1.Application{}
+	if err := r.Client.Get(ctx, types.NamespacedName{Name: applicationName, Namespace: namespace}, application); err != nil {
+		log.Error(err, "failed to get Application", "application", applicationName)
+		return err
+	}
+
+	applicationPullSecretName := getApplicationPullSecretName(applicationName)
+	applicationPullSecret := &corev1.Secret{}
+	if err := r.Client.Get(ctx, types.NamespacedName{Name: applicationPullSecretName, Namespace: namespace}, applicationPullSecret); err != nil {
+		log.Error(err, "failed to get application pull secret", "secretName", applicationPullSecretName)
+		return err
+	}
+
+	pullSecret := &corev1.Secret{}
+	if err := r.Client.Get(ctx, types.NamespacedName{Name: pullSecretName, Namespace: namespace}, pullSecret); err != nil {
```

Comment:
Do we really need to get the secret every time?

Reply:
well we need it's content to compare it with application secret